src/Eloquent/DocumentModel.php

```diff
@@ -97,8 +99,14 @@ public function getQualifiedKeyName()
         return $this->getKeyName();
     }
 
-    /** @inheritdoc */
-    public function fromDateTime($value)
+    /**
+     * Convert a DateTime to a storable UTCDateTime.
```

Comment:
Should this refer to DateTimeInterface? You can also write "DateTimeInterface (including Carbon)" for more clarity.

Related thoughts (no action needed) follow.

Although UTCDateTime accepts a DateTimeInterface, it doesn't actually rely on any of its methods. That interface is effectively used in place of a union on DateTime and DateTimeImmutable. The [UTCDateTime constructor](https://github.com/mongodb/mongo-php-driver/blob/1.19.3/src/BSON/UTCDateTime.c#L151) accesses the internal `php_date_obj` struct directly.

Thankfully, this is safe because [DateTimeInterface](https://www.php.net/manual/en/class.datetimeinterface.php) is not able to be implemented by userland classes. So any instance will be either a DateTime or DateTimeImmutable object with the internal struct.